File: html/shared/debug.js
// ============================================
// FW Core - Zentrales Debug-System (NUI/Frontend)
// ============================================
// Verwendung:
//   FWDebug.log('Inventory', 'Item added', itemName, amount)
//   FWDebugTable('Player', 'State', stateObject)
// ============================================

window.FWDebug = (function() {
    let debugEnabled = false;
    
    // Lade Debug-Config vom Client
    const loadDebugConfig = () => {
        window.NUIBridge.send('getDebugConfig', {}, (enabled) => {
            debugEnabled = enabled;
            if (debugEnabled) {
                console.log('[FW Debug] 🐛 Frontend Debug-Modus aktiviert');
            }
        });
    };
    
    // Initial laden
    loadDebugConfig();
    
    // Debug Log mit Kategorie
    const debug = (category, message, ...args) => {
        if (!debugEnabled) return;
        
        const prefix = `[FW:${category}]`;
        if (args.length > 0) {
            console.log(prefix, message, '|', ...args);
        } else {
            console.log(prefix, message);
        }
    };
    
    // Debug Table (für Objekte)
    const debugTable = (category, message, obj) => {
        if (!debugEnabled) return;
        
        console.log(`[FW:${category}] ${message}:`, obj);
    };
    
    // Debug Toggle (für Live-Änderung)
    const setDebug = (enabled) => {
        debugEnabled = enabled;
        console.log('[FW Debug] Debug-Modus:', enabled ? 'AN' : 'AUS');
    };
    
    return {
        log: debug,
        table: debugTable,
        setDebug: setDebug,
        isEnabled: () => debugEnabled
    };
})();

// Globale Shortcuts
window.FWDebug.log = window.FWDebug.log;
window.FWDebugTable = window.FWDebug.table;

File: shared/debug.lua
-- ============================================
-- FW Core - Zentrales Debug-System
-- ============================================
-- Verwendung:
--   FW.Debug('Inventory', 'Item added', itemName, amount)
--   FW.DebugTable('Player', 'Inventory state', inventoryTable)
-- ============================================

FW = FW or {}

-- Lade Debug-Config aus main.json
local function LoadDebugConfig()
    local resourceName = GetCurrentResourceName()
    local content = LoadResourceFile(resourceName, 'configs/main.json')
    
    if content then
        local ok, data = pcall(json.decode, content)
        if ok and data and data.debug ~= nil then
            Config.Debug = data.debug
            return
        end
    end
    
    -- Fallback
    Config.Debug = false
end

-- Beim Resource-Start laden
CreateThread(function()
    LoadDebugConfig()
    if Config.Debug then
        print('[FW Debug] 🐛 Debug-Modus aktiviert')
    end
end)

-- Debug Print mit Kategorie
function FW.Debug(category, message, ...)
    if not Config or not Config.Debug then return end
    
    local args = {...}
    local argsStr = ''
    
    if #args > 0 then
        local argParts = {}
        for i, v in ipairs(args) do
            if type(v) == 'table' then
                table.insert(argParts, json.encode(v))
            else
                table.insert(argParts, tostring(v))
            end
        end
        argsStr = ' | ' .. table.concat(argParts, ', ')
    end
    
    print(string.format('[FW:%s] %s%s', category, message, argsStr))
end

-- Debug Table Print (für komplexe Datenstrukturen)
function FW.DebugTable(category, message, tbl)
    if not Config or not Config.Debug then return end
    
    if type(tbl) == 'table' then
        print(string.format('[FW:%s] %s:', category, message))
        print(json.encode(tbl, {indent = true}))
    else
        FW.Debug(category, message, tbl)
    end
end

-- Export für Client/Server
if IsDuplicityVersion() then
    -- Server-Side
    print('[FW Debug] 📡 Server debug helper loaded')
else
    -- Client-Side
    print('[FW Debug] 💻 Client debug helper loaded')
end
